probatus/utils/shap_helpers.py

```diff
@@ -62,9 +60,18 @@ def shap_calc(model, X, approximate=False, return_explainer=False, verbose=0, **
         if verbose <= 100:
             warnings.simplefilter("ignore")
 
-        explainer = shap.TreeExplainer(model, **shap_kwargs)
-        # Calculate Shap values
-        shap_values = explainer.shap_values(X, approximate=approximate)
+        # Create the background data,required for non tree based models.
+        # A single datapoint can passed as mask (https://github.com/slundberg/shap/issues/955#issuecomment-569837201)
+        
+        if X.shape[1]< sample_size :
+            sample_size = int(np.ceil(X.shape[1]*0.2))
+        else :
+           pass
+        mask = shap.utils.sample(X,sample_size)
+
+        explainer = shap.Explainer(model,masker=mask,**shap_kwargs)
```

Comment:
Isn't this step a bit dangerous?
If we pass a tree based model, this would still use the Explainer (instead of the Tree Explainer), which is not something you really want, right?

Reply:
I think the new shap basically selects the explainer to be used based on the provided model.

If you pass tree model it will use tree explainer within Explainer

Reply:
With the latest version of SHAP ,[Explainer](https://shap.readthedocs.io/en/latest/generated/shap.Explainer.html#shap.Explainer.__init__) by default attempts to make the best choice given the passed model. So this spares identifying the model type in the code.
However,  having said that I think models like SVC(kernel) and KNN etc needs to handled separately.